php-checks/src/main/java/org/sonar/php/checks/security/HttpRequestCheck.java

```diff
@@ -0,0 +1,123 @@
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.checks.utils.namespace.NamespaceAwareVisitor;
+import org.sonar.php.checks.utils.namespace.QualifiedName;
+import org.sonar.plugins.php.api.tree.SeparatedList;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.tree.expression.NewExpressionTree;
+
+import static org.sonar.php.checks.utils.namespace.QualifiedName.qualifiedName;
+
+@Rule(key = "S4825")
+public class HttpRequestCheck extends NamespaceAwareVisitor {
+
+  private static final String MESSAGE = "Make sure that this http request is sent safely.";
+
+  private static final List<QualifiedName> SUSPICIOUS_CLASS_INSTANTIATIONS = Arrays.asList(
+    qualifiedName("http\\Client\\Request"),
+    qualifiedName("GuzzleHttp\\Client"));
+
+  private static final List<FunctionMatcher> IO_FUNCTIONS = Arrays.asList(
+    new FunctionMatcher(QualifiedName.create("copy"), 0, 1),
+    new FunctionMatcher(QualifiedName.create("curl_exec")),
+    new FunctionMatcher(QualifiedName.create("file"), 0),
+    new FunctionMatcher(QualifiedName.create("file_get_contents"), 0),
+    new FunctionMatcher(QualifiedName.create("fopen"), 0),
+    new FunctionMatcher(QualifiedName.create("readfile"), 0),
+    new FunctionMatcher(QualifiedName.create("get_headers")),
+    new FunctionMatcher(QualifiedName.create("get_meta_tags"), 0));
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    if (isSuspiciousGlobalFunction(tree) || isSuspiciousClassInstantiation(tree.callee())) {
+      context().newIssue(this, tree, MESSAGE);
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  @Override
+  public void visitNewExpression(NewExpressionTree tree) {
+    if (isSuspiciousClassInstantiation(tree.expression())) {
+      context().newIssue(this, tree, MESSAGE);
+    }
+    super.visitNewExpression(tree);
+  }
+
+  private boolean isSuspiciousClassInstantiation(ExpressionTree expression) {
+    if (expression.is(Tree.Kind.NAMESPACE_NAME)) {
+      NamespaceNameTree namespaceName = (NamespaceNameTree) expression;
+      QualifiedName className = getFullyQualifiedName(namespaceName);
+      return SUSPICIOUS_CLASS_INSTANTIATIONS.stream().anyMatch(className::equals);
+    }
+    return false;
+  }
+
+  private boolean isSuspiciousGlobalFunction(FunctionCallTree tree) {
+    ExpressionTree callee = tree.callee();
+    if (callee.is(Tree.Kind.NAMESPACE_NAME)) {
+      NamespaceNameTree namespaceName = (NamespaceNameTree) callee;
+      QualifiedName qualifiedName = getFullyQualifiedName(namespaceName);
+      return IO_FUNCTIONS.stream().anyMatch(matcher -> matcher.matches(qualifiedName, tree.arguments()));
```

Comment:
I would extract `tree.arguments()` into a dedicated variable, in order to avoid call to the method for all the functions.